<a href="https://colab.research.google.com/github/ThomassEscobar/Medias-de-dispersion/blob/main/Laboratorio_5_Medidas_de_Dispersi%C3%B3n.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Medidas de Dispersión

## Objetivos:

Aplicar medidas de dispersión (rango, desviación estándar y coeficiente de variación) en el análisis de bases de datos, utilizando Python como herramienta de apoyo.

*Nota: Este laboratorio está diseñado para desarrollar razonamiento estadístico. Evita el uso de herramientas de IA para responder las preguntas a menos que esto sea explícitamente sugerido* 🤖. *En su lugar, enfócate en aplicar los conceptos revisados.*

## Actividad 1

El gobierno de la ciudad de San Francisco ha publicado los sueldos base de sus funcionarios públicos entre los años 2011 y 2014. Esta base incluye trabajadores con distintos cargos y niveles de responsabilidad.
Como analista de datos, tu tarea es describir y cuantificar la variabilidad de los sueldos base usando medidas estadísticas de dispersión.

**Utiliza la siguiente base de datos para responder las actividades 1, 2 y 3.**

Descarga la base de datos **<a href="https://drive.google.com/uc?export=download&id=1Ry63iApbDJaY_C1HfR5s-kIX5ldWcGW8
">aquí</a>**.

*Observación: Considera que la variable Pago base está expresada en dólares estadounidenses (USD) y corresponde a ingresos anuales.*



1. Identifica la población y muestra del estudio.

In [1]:
import pandas as pd
df_1 = pd.read_csv('Actividad_1_2_3.csv')
df_1.shape[0]



2270

Poblacion: Los funcionarios publicos de la ciudad de San Francisco entre los años 2011 y 2014.

Muestra: 2270

2. Muestra las tres primeras y tres últimas filas de la base de datos.

In [ ]:
df_1.head(3)

In [ ]:
df_1.tail(3)

3. Calcula el rango de la variable *Pago base*, ¿este valor representa la realidad de todos los trabajadores?

In [2]:
max_pb = df_1['Pago base'].max()
min_pb = df_1['Pago base'].min()
rango = max_pb - min_pb
print(f'El rango de los sueldos de los trabajores es de ${rango}[USD]')

El rango de los sueldos de los trabajores es de $272497[USD]


Este valor indica que existe una gran diferencia eltre el sueldo mas alto y el sueldo mas bajo

4. Calcula la media, mediana, desviación estándar y coeficiente de variación del sueldo base. ¿La dispersión es alta o baja respecto al promedio? ¿Qué medida (media o mediana) consideras más representativa para describir el sueldo típico? Justifica tu respuesta.



In [4]:
media_pb = df_1['Pago base'].mean()
mediana_pb = df_1['Pago base'].median()
desvest_pb = df_1['Pago base'].std()

cv_pb = (desvest_pb/media_pb)*100

print(f'La media de los sueldos es de ${media_pb:.0f}[USD]')
print(f'La mediana de los sueldos es de ${mediana_pb:.0f}[USD]')
print(f'La desviacion estandar de los sueldos es de ${desvest_pb:.0f}[USD]')
print(f'El coeficiente de variacion de los sueldos es de {cv_pb:.1f}%')




La media de los sueldos es de $77087[USD]
La mediana de los sueldos es de $74712[USD]
La desviacion estandar de los sueldos es de $31122[USD]
El coeficiente de variacion de los sueldos es de 40.4%


Dado los resultados diremos que los sueldos presentan una alta dispersion, por lo mismo, consideraremos la mediana como un valor mas representativo.

## Actividad 2

Un equipo de análisis de datos quiere comparar la variabilidad de los sueldos base entre distintos años para evaluar si ha habido más o menos estabilidad en las remuneraciones del empleo público.

Trabajarás con la base de datos que ha publicado el gobierno de San Francisco con información de los sueldos base de sus funcionarios públicos entre los años 2011 y 2014. Esta base incluye trabajadores con distintos cargos y niveles de responsabilidad.

**Utiliza la siguiente base de datos para responder las actividades 1, 2 y 3.**

Descarga la base de datos **<a href="https://drive.google.com/uc?export=download&id=1Ry63iApbDJaY_C1HfR5s-kIX5ldWcGW8
">aquí</a>**.

*Observación: Considera que la variable Pago base está expresada en dólares estadounidenses (USD) y corresponde a ingresos anuales.*

1. Importa la base de datos y crea una tabla que agrupe los valores de *Pago base*. Luego, calcula la media, la desviación estándar y el coeficiente de variación por año.

In [5]:
import pandas as pd

df_2 = pd.read_csv('Actividad_1_2_3.csv')

In [6]:
tabla_2 = df_2.groupby('Año')['Pago base'].agg(['mean','std'])
tabla_2['cv [%]'] = (tabla_2['std']/tabla_2['mean'])*100
round(tabla_2, 1)

,mean,std,cv [%]
Año,,,
2011,79206.6,26326.9,33.2
2012,74188.2,33407.9,45.0
2013,76963.0,32361.8,42.0
2014,89173.0,42143.0,47.3


2. ¿En cuál año se observa mayor dispersión absoluta en los sueldos base? Justifica usando la desviación estándar.

3. ¿Cuál año tiene mayor heterogeneidad? Justifica usando el coeficiente de variación.

4. ¿Qué ventajas tiene el coeficiente de variación frente a la desviación estándar al realizar comparaciones?

5. ¿Qué interpretación puedes hacer al comparar los datos del año 2011 con los del 2014?

## Actividad 3

Un equipo de análisis de datos busca estudiar qué tan parecidos o distintos son los sueldos base entre trabajadores que ejercen el mismo cargo. Para ello, se analizarán los cinco títulos profesionales más frecuentes en la base de datos publicada por el gobierno de San Francisco, la cual contiene información de los sueldos base de funcionarios públicos entre los años 2011 y 2014. Esta base incluye trabajadores con distintos niveles de responsabilidad.

Como analista, tu tarea es calcular y comparar medidas estadísticas de dispersión dentro de cada cargo, para evaluar la variabilidad de las remuneraciones en estos grupos.

**Utiliza la siguiente base de datos para responder las actividades 1, 2 y 3.**

Descarga la base de datos **<a href="https://drive.google.com/uc?export=download&id=1Ry63iApbDJaY_C1HfR5s-kIX5ldWcGW8
">aquí</a>**.

*Observación: Considera que la variable Pago base está expresada en dólares estadounidenses (USD) y corresponde a ingresos anuales.*

1. Crea una tabla de frecuencias absolutas que muestre los cuatro cargos  más frecuentes de San Francisco, usando la variable *Título profesional*.

2. ¿Qué observas respecto a los cargos más comunes? ¿Te sorprende alguno?

3. De acuerdo con la tabla realizada en el inciso anterior, crea un DataFrame que contenga solo los cuatro cargos con mayor presencia en la ciudad de San Francisco, para esto debes filtrar tu DataFrame original. Luego, muestra las cinco primeras filas y las cinco últimas filas.

$\hspace{1cm}$ *Para filtrar rápidamente puedes almacenar los cargos frecuentes en una variable y luego utilizar la función isin.*

4. Utiliza el DataFrame filtrado del inciso anterior para crear una tabla con el cálculo de la media, mediana, desviación estándar y coeficiente de variación del sueldo base (*Pago base*).

5. ¿Cuál de los cargos tiene sueldos más homogéneos? Justifica.